tests/include/mir/test/doubles/stub_buffer_allocator.h

```diff
@@ -38,7 +38,7 @@ namespace doubles
 
 namespace
 {
-void memcpy_from_shm_buffer(struct wl_shm_buffer* buffer)
```

Comment:
How is that a possible ODR violation? The anon namespace ensures that different translation units have distinct definitions.

Reply:
Oh! To answer my own question: it is used by an inline function in the header. Different translation units will have different definitions of the `buffer_from_shm()` function as they use different definitions of `buffer_from_shm()`.

But inlining `buffer_from_shm()` doesn't fix that. (Even if it placates clang-tidy)